src/conta/tests/conta.service.spec.ts

```diff
@@ -1,18 +1,140 @@
+    (clienteRepository.lerClientes as jest.Mock).mockReturnValue(clientes);
+    (contaFactory.criarConta as jest.Mock).mockReturnValue(novaConta);
+
+    const resultado = contaService.criarConta(saldo, clienteId, tipoConta);
+
+    expect(resultado).toEqual(novaConta);
+    expect(contaRepository.lerContas).toHaveBeenCalled();
+    expect(clienteRepository.lerClientes).toHaveBeenCalled();
+    expect(contaFactory.criarConta).toHaveBeenCalledWith(
+      1,
+      saldo,
+      clienteId,
+      tipoConta,
+    );
+    expect(contaRepository.escreverContas).toHaveBeenCalledWith([novaConta]);
+  });
+
+  test('Deve retornar NotFoundException ao enviar um clientId inexistente', () => {
+    (contaRepository.lerContas as jest.Mock).mockReturnValue(contas);
+    (clienteRepository.lerClientes as jest.Mock).mockReturnValue(clientes);
+    (contaFactory.criarConta as jest.Mock).mockReturnValue(novaConta);
+
+    clienteId = 12345;
+
+    expect(() => contaService.criarConta(saldo, clienteId, tipoConta)).toThrow(
+      NotFoundException,
+    );
+  });
+
+  test('Deve alterar o tipo da conta', () => {
+    (contaRepository.lerContas as jest.Mock).mockReturnValue(contas);
+    (clienteRepository.lerClientes as jest.Mock).mockReturnValue(clientes);
+    (contaFactory.criarConta as jest.Mock).mockReturnValue(novaConta);
+
+    clienteId = 1;
+    contaService.criarConta(saldo, clienteId, tipoConta);
+    const resultado = contaService.modificarTipoConta(1, TipoConta.POUPANCA);
+    const esperado = new Conta(1, saldo, clienteId, TipoConta.POUPANCA);
+
+    expect(resultado).toStrictEqual(esperado);
+    expect(contaRepository.lerContas).toHaveBeenCalled();
+    expect(contaRepository.escreverContas).toHaveBeenCalled();
+  });
+
+  test('Deve retornar NotFoundException se conta não for encontrada', () => {
+    (contaRepository.lerContas as jest.Mock).mockReturnValue(contas);
+    (clienteRepository.lerClientes as jest.Mock).mockReturnValue(clientes);
+    (contaFactory.criarConta as jest.Mock).mockReturnValue(novaConta);
+
+    expect(() =>
+      contaService.modificarTipoConta(12345, TipoConta.POUPANCA),
+    ).toThrow(NotFoundException);
+  });
+
+  test('Deve remover a conta solicitada', () => {
+    (contaRepository.lerContas as jest.Mock).mockReturnValue(contas);
+    (clienteRepository.lerClientes as jest.Mock).mockReturnValue(clientes);
+    (contaFactory.criarConta as jest.Mock).mockReturnValue(novaConta);
+
+    clienteId = 1;
+    contaService.criarConta(saldo, clienteId, tipoConta);
+    const resultado = contaService.removerConta(1);
+
+    // eslint-disable-next-line prettier/prettier
+    expect(resultado).toStrictEqual({ 'message': 'Conta deletada com sucesso!' });
+    expect(contas).toHaveLength(0);
+    expect(contaRepository.lerContas).toHaveBeenCalled();
+    expect(contaRepository.escreverContas).toHaveBeenCalled();
+  });
+
+  test('Deve retornar NotFoundException se conta não for encontrada', () => {
```

Comment:
Pra não ficar com múltiplos testes com o mesmo nome, você poderia colocar a qual função se refere o teste no nome (por exemplo 'Deve retornar NotFoundException se conta não for encontrada ao tentar remover conta'), ou pode colocar blocos describe() dentro do principal pra agrupar os testes de cada função.